src/dialer/index.js

```diff
@@ -120,7 +124,16 @@ class Dialer {
       this.peerStore.addressBook.add(id, multiaddrs)
     }
 
-    let knownAddrs = this.peerStore.addressBook.getMultiaddrsForPeer(id) || []
+    let knownAddrs = this.addressSorter(
+      this.peerStore.addressBook.get(id) || []
```

Comment:
Any reason you switched away from `getMultiaddrsForPeer`? The address sorter should be able to handle `/p2p` on a multiaddr and avoids the need for the map logic here.

Reply:
Yes! I want the sorter to get an `Address` and not a `multiaddr`. With this, we have the ability to sort with other data, like the `.isCertified` boolean.

Reply:
The downside is having to do the `/p2p` logic which is embedded in the `getMultiaddrsForPeer`. I suggest I move that logic to `libp2p-utils` and reuse it here. What do you think? Any naming suggestions?

Reply:
Ah, right. What if we just extend `getMultiaddrsForPeer` to take an optional sorter? Then we can keep that logic isolated and allow different systems to use different sorters as needed.

Reply:
that seems good! I will change that as we will only use this with the `getMultiaddrsForPeer`